### ISSUE TITLE
Raspberry Pi 16 Channel Relay not working correctly 

### ISSUE THREAD

**Issue by engine-ad-oz** [skiwithpete/relaypi#5]
Hi Pete,

I followed your youtube video and downloaded the python scripts. I've wired the relay board and raspberry Pi as per your video. The problem I'm having is I can't cycle the relays as your video does. They come on together and switch off together.

I'm sure the code is doing the correct thing and I'm confident the wiring is ok. Are there any subtleties with the relay boards? I got this one off ebay, there are no jumpers to set or anything, just connect 12v and the pi and I think it should work.

Anyway cool videos and very well presented.

Hope to hear from you soon.

Cheers

engine-ad-oz

**Comment by skiwithpete** [comment]
can you upload a pic of it?  I worry its not an optocoupler relay...

**Comment by engine-ad-oz** [comment]
[Image]



Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by skiwithpete** [comment]
sorry, i don't see an image.  just the word image.

post a link from imgur

**Comment by engine-ad-oz** [comment]
[Image]
________________________________

**Comment by skiwithpete** [comment]
Nope, sorry still only see [image]

**Comment by engine-ad-oz** [comment]
![thumbnail_20200426_154127](https://user-images.githubusercontent.com/64311075/80461666-f94ee700-8978-11ea-99c6-3fdd55f30fc4.jpg)

Hope that comes through.

**Comment by skiwithpete** [comment]
Yep, that's what an optocoupler relay looks like, so that's not the issue...

Umm... When you plug in just two relays.  You can't get them to fire
individually?

**Comment by engine-ad-oz** [comment]
Yes, they all turn on and off at the same time. There's no individual control.

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by skiwithpete** [comment]
what I'm asking is: even if you only plug two in?  Do they all come on?

**Comment by engine-ad-oz** [comment]
Sorry misunderstood, only the ones that are connected. So in that example only 2 would come on but with no independent control.

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by skiwithpete** [comment]
when they click on what is your script doing?  Are you able to control
them?  Like you can get them to switch on and off?  Just both at the same
time?

**Comment by engine-ad-oz** [comment]
Script does what it says it should. E.g. script 1 will print 'One, Two,' etc the relays will switch off when it gets to the end

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by skiwithpete** [comment]
This is so highly unusual.  I mean, I've had this script up for years, and
been dealing with issues across that time, and I haven't heard of this
before.

Two possibilities
1) incorrect wiring.
2) defective unit.

Let's look at 1, before we jump to 2.  There are ways to power the 16
channel relay with just two switches that relies on 5V rather than the full
12V.  Have you tried that?  Or, did you make any modifications to the
code?  IE the pin numbers?

That's pretty much it... like I've gone through all the things that
typically go wrong with these things, and If that still doesn't work, I've
got to think you've got a defective switch board.  One that's leaking
across the line.  So that powering one does multiples... but even then,
that typically when you only power 2 switches, it would still light up all
16.  So I'm really stumped here.

Maybe one last last last thing.  You've powered up 2 switches, I assume you
chose 1&2 can you try 2&3 or 3&4?

And for good measure - now that we've taken pics.  Can you take a pic of
your wiring for your 2 test switches as above?

**Comment by engine-ad-oz** [comment]
I'm guessing its a faulty board, the pi works fine for everything else I play around with and the only code changes I made were to rem out lines that werent required as I cut down the number of relays I was trying.

I'll take some pics of the wiring when I het a chance, ive got another relay board coming and another pi to try too.

Thanks for you help.

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by engine-ad-oz** [comment]
Hi Pete,

Just wanted to let you know it was a fault on the relay board. I got another one and that works fine. Thanks for the support.

Cheers

Ad
